testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -162,6 +162,22 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
         assert_equal(results, ana.results)
 
 
+def test_AnalysisFromFunction_args_order(u):
+    # Issue 2503
+    def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+    
+    protein = u.select_atoms('protein')
+    ans1 = base.AnalysisFromFunction(mass_xyz, u.trajectory, protein, protein.masses.reshape(-1, 1))
```

Comment:
In my opinion, you don't need to have `ans1` here, since this test is for when the trajectory is not given. However, the tests don't actually have a case for when the trajectory is given, so it would be great if you made this a separate test and checked the argument order, and `ans1._trajectory` identity, again.

Reply:
what do you mean by `made this a separate test and checked the argument order`, one function for testing case and another for testing argument order?

Reply:
As @orbeckst elaborated on below, yes, checking the arguments when a trajectory is not given should be done in a separate test to checking the arguments when a trajectory is given.